orangecontrib/spectroscopy/widgets/owhyper.py

```diff
@@ -1119,9 +1260,17 @@ def image_values(self):
             else:
                 return Integrate(methods=imethod,
                                  limits=[[self.choose, self.choose]])
-        else:
+        elif self.value_type == 1:  # feature
             return lambda data, attr=self.attr_value: \
                 data.transform(Domain([data.domain[attr]]))
+        elif self.value_type == 2:  # RGB
+            # Ensure all RGB variables are unique
```

Comment:
Why testing for uniqueness here? Is there any harm in drawing values directly, even if not unique? There might be a speed penalty, but drawing happens in a separate thread that can be cancelled anytime (by choosing other variables, for instance) so this is a non-issue.

If we wanted to enforce uniqueness, we would need to also warn and not just eat the choice. But I'd actually prefer to repeat. Then could make a perfectly controlled grayscale image from a certain feature by repeating it 3 times. :)

Reply:
We are also fine with re-using attributes but `Domain([attr, attr])` raises an exception if `attr == attr` (See L153 of `Orange.data.domain`)
If there's another way to tackle this and drop both the uniqueness test and the None return (see above) I'm all for it